packages/core/src/odata/common/entity-deserializer.ts

```diff
@@ -19,7 +19,7 @@ import { CollectionField } from '../v4/selectable/collection-field';
 /**
  * @experimental This is experimental and is subject to change. Use with caution.
  */
-export function entityDeserializer(edmToTs) {
+export function entityDeserializer(edmToTs, extractODataETag) {
```

Comment:
Minor: Instead of passing the parser for v2/v4 explicitly you could also extract the `_oDataVersion` from the entity and from there use a factory to get the right extract function. However, I guess for the currently two function it is fine and `edmToTs` is the other v2/v4 dependent thingy.